homeassistant/setup.py

```diff
@@ -181,6 +184,7 @@ def log_error(msg: str, link: Optional[str] = None) -> None:
             )
         else:
             log_error("No setup function defined.")
+            hass.data[DATA_SETUP_STARTED].pop(domain)
```

Comment:
We can just put this in the `finally` ?

Reply:
Oh right then we would miss out on config entries.

Reply:
Exactly.  I tried that first